src/ModuleRegistry.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.21;
+
+import { Module } from "./struct/Module.sol";
+import "openzeppelin-contracts/contracts/proxy/utils/Initializable.sol";
+import "openzeppelin-contracts/contracts/utils/introspection/ERC165Checker.sol";
+
+/**
+ * @title Module Registry
+ * @author Consensys
+ * @notice This contract aims to manage the Modules used by the Portals, including their discoverability
+ */
+contract ModuleRegistry is Initializable {
+  /// @dev The contract administrator
+  address private admin;
+  /// @dev The list of Modules, accessed by their ID
+  mapping(address id => Module module) public modules;
+  /// @dev The number of Modules already registered
+  uint256 public numberOfModules = 0;
```

Comment:
No need, in #41 we'll add an array of modules address and read its length to get this value